### DISCUSSION TITLE
[C Wrapper] How to access fields of records from a DjinniRecordHandle?

### DISCUSSION THREAD

**Discussion by gibutm** [cross-language-cpp/djinni-generator#140] [Q&A]
I'm trying to consume the generated c wrappers for consumption with dart.

If i define a record `R` and an interface `I` which contains a method `M` that returns an instance of R. 
The generated C wrapper outputs look like 
I -> struct DjinniWrapperI 
R -> struct DjinniR
M-> struct DjinniRecordHandle * cw__I_get_R_instance(struct DjinniWrapperI * djinni_this);

The goal is to access 2 member fields that exist inside R once i've invoked M
Definition of R looks like this.
```
R = record {
	width: i32;
	height: i32;
} 
```

Question 1: How do you get `DjinniR *` pointer from `DjinniRecordHandle *`. Would casting the pointer be the right approach?
Question 2: How can i access the fields `width` and `height` with an instance of `DjinniR *` ?

**Comment by paulocoutinhox** [comment]
Same examples here about accessing struct fields:

- https://dart.dev/guides/libraries/c-interop
- https://api.dart.dev/stable/2.18.3/dart-ffi/Struct-class.html
- https://medium.com/dartlang/implementing-structs-by-value-in-dart-ffi-1cb1829d11a9
- https://groups.google.com/g/dart-ffi/c/krGzNp1HnwA
